eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -45,10 +47,26 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
   val id = cfg.id
   val paymentsDb = nodeParams.db.payments
 
+  private val span = Kamon.runWithContextEntry(MultiPartPaymentLifecycle.parentPaymentIdKey, cfg.parentId) {
+    val spanBuilder = if (Kamon.currentSpan().isEmpty) {
+      Kamon.spanBuilder("single-payment")
+    } else {
+      Kamon.spanBuilder("payment-part").asChildOf(Kamon.currentSpan())
+    }
+    spanBuilder
+      .tag("paymentId", cfg.id.toString)
+      .tag("paymentHash", cfg.paymentHash.toHex)
+      .tag("targetNodeId", cfg.targetNodeId.toString())
+      .start()
+  }
+
   startWith(WAITING_FOR_REQUEST, WaitingForRequest)
 
   when(WAITING_FOR_REQUEST) {
     case Event(c: SendPaymentToRoute, WaitingForRequest) =>
+      span.tag("amount", c.finalPayload.amount.toLong)
```

Comment:
@t-bast we could avoid repetition here by making `SendPaymentToRoute` and `SendPayment` derive from a common trait. Curious why you didn't do it?

Reply:
I didn't change it because it was already in the code, with the two types separate, but that can probably be done yeah ;)